torchtitan/config_manager.py

```diff
@@ -480,18 +480,17 @@ def __init__(self):
             help="Flight recorder ring buffer size, >0 means recording by default, 0 means disabled",
         )
 
-        # estimation mode settings
+        # memory estimation settings
         self.parser.add_argument(
-            "--estimate.memory",
+            "--memory_estimation.enabled",
             help="Whether to estimate memory usage for FSDP",
-            default=False,
+            action="store_true",
         )
 
         self.parser.add_argument(
-            "--estimate.mode",
-            type=str,
-            default="fake",
-            help="Mode of estimation to use ['fake', 'real']",
+            "--memory_estimation.fake_mode_only",
+            help="Whether to estimate memory under FakeTensorMode",
+            action="store_true",
```

Comment:
If you don't specify default value, default value would be `False` because of `store_true`. Let's specify default value to be `True`.

Reply:
@tianyu-l We do want the value of the flag to be `False` if the user hasn't specified `--memory_estimation.fake_mode_only` flag? If we want the the `FakeTensorMode` to be active by default, we should rename this flag as `--memory_estimation.fake_mode_disabled`, so that users only use if if they want to disable fake mode.